ios/FlankerNativeComponents/GameManager.swift

```diff
@@ -184,271 +124,257 @@ class GameManager {
-    }
+      text = gameParameters.trials[countTest].stimulus.en
 
-    timeResponse = Timer(timeInterval: gameParameters.trialDuration / 1000, target: self, selector: #selector(self.timeResponseFailed), userInfo: nil, repeats: false)
-    RunLoop.main.add(timeResponse!, forMode: .common)
+      if let image = URL(string: text), text.contains("https") {
+          delegate?.updateFixations(image: image, isStart: true, typeTime: .trial)
+      } else {
+          delegate?.updateText(text: text, color: .black, font: Constants.bigFont, isStart: true, typeTime: .trial)
+      }
+
+      DispatchQueue.main.asyncAfter(deadline: .now() + 0.05) {
+          self.delegate?.setEnableButton(isEnable: true)
+          self.timeResponse = Timer(timeInterval: gameParameters.trialDuration / 1000, target: self, selector: #selector(self.timeResponseFailed), userInfo: nil, repeats: false)
+          RunLoop.main.add(self.timeResponse!, forMode: .common)
+      }
   }
 
   @objc func timeResponseFailed() {
     guard let gameParameters = gameParameters else { return }
 
     delegate?.setEnableButton(isEnable: false)
+
+    endTrialTimestamp = CACurrentMediaTime()
+    setEndTimeViewingImage(time: endTrialTimestamp!, isStart: false, type: .trial)
+
+    startFeedbackTimestamp = CACurrentMediaTime()
+    setEndTimeViewingImage(time: startFeedbackTimestamp!, isStart: true, type: .feedback)
+
     if gameParameters.showFeedback {
-      delegate?.updateText(text: Constants.timeRespondText, color: .black, font: Constants.smallFont, isStart: false, typeTime: .feedback)
+        delegate?.updateText(text: Constants.timeRespondText, color: .black, font: Constants.smallFont, isStart: false, typeTime: .feedback)
     }
 
-    guard
-      let startTrialTimestamp = startTrialTimestamp,
-      let endTrialTimestamp = endTrialTimestamp
-    else { return }
-
-    let model = FlankerModel(rt: 0.0,
-                             stimulus: text,
-                             button_pressed: nil,
-                             image_time: endTrialTimestamp * 1000, // має намалювати
-                             correct: false,
-                             start_timestamp: 0, // вже намальовано
-                             tag: Constants.tag,
-                             trial_index: countTest + 1,
-                             start_time: startTrialTimestamp * 1000,
-                             response_touch_timestamp: 0)
+    guard let startTrialTimestamp = startTrialTimestamp else { return }
+
+    let model = FlankerModel(
+        rt: 0.0,
+        stimulus: text,
+        button_pressed: nil,
+        image_time: endTrialTimestamp! * 1000, // має намалювати
+        correct: false,
+        start_timestamp: 0, // вже намальовано
+        tag: Constants.tag,
+        trial_index: countTest + 1,
+        start_time: startTrialTimestamp * 1000,
+        response_touch_timestamp: 0
+    )
```

Comment:
There's a lot of indentation issues in all the changes.

```suggestion
    let model = FlankerModel(
      rt: 0.0,
      stimulus: text,
      button_pressed: nil,
      image_time: endTrialTimestamp! * 1000,  // має намалювати
      correct: false,
      start_timestamp: 0,  // вже намальовано
      tag: Constants.tag,
      trial_index: countTest + 1,
      start_time: startTrialTimestamp * 1000,
      response_touch_timestamp: 0
    )
```